script.js

```diff
@@ -1,4 +1,204 @@
+  var autoCombinedNum = 0;
+  var indexArray = input.split("").map(Number);
+  // This loop takes in the user input sequence and generate the combined number.
+  for (var i = 0; i < indexArray.length; i += 1) {
+    combinedNum = combinedNum * 10 + diceArray[indexArray[i]];
+  }
+  // This loop auto generates the highest/lowest number depending on the game version.
+  var index = 0;
+  var maxIndex = 0;
+  var maxNum = 0;
+  var tempArray = diceArray;
+  while (autoCombinedNum.toString().length <= diceArray.length + 1) {
+    if (gameVersion == "highest") {
+      while (index < tempArray.length) {
+        if (tempArray[index] > maxNum) {
+          maxNum = tempArray[index];
+          maxIndex = index;
+          index += 1;
+        } else {
+          index += 1;
+        }
+      }
+      autoCombinedNum = autoCombinedNum * 10 + maxNum;
+      tempArray.splice(maxIndex, 1);
+      maxNum = 0;
+      index = 0;
+    } else {
+      maxNum = 7;
+      while (index < tempArray.length) {
+        if (tempArray[index] < maxNum) {
+          maxNum = tempArray[index];
+          maxIndex = index;
+          index += 1;
+        } else {
+          index += 1;
+        }
+      }
+      autoCombinedNum = autoCombinedNum * 10 + maxNum;
+      tempArray.splice(maxIndex, 1);
+      maxNum = 7;
+      index = 0;
+    }
+  }
+
+  return [combinedNum, autoCombinedNum];
+};
+
+// Compare the numbers of the players and declare the winning player. Function takes in the array of combined players numbers. Returns the highest number and the winning player.
+var compareNumbers = function (input) {
+  var maxNum = input[0];
+  var maxIndex = 0;
+  var index = 0;
+  if (gameVersion == "highest") {
+    while (index < input.length) {
+      if (input[index] > maxNum) {
+        maxNum = input[index];
+        maxIndex = index;
+        index += 1;
+      } else {
+        index += 1;
+      }
+    }
+    playerCount = 0;
+  } else {
+    while (index < input.length) {
+      if (input[index] < maxNum) {
+        maxNum = input[index];
+        maxIndex = index;
+        index += 1;
+      } else {
+        index += 1;
+      }
+    }
+    playerCount = 0;
+  }
+  return [maxNum, maxIndex + 1];
+};
+
+// Show leaderboard. Function takes in array of running scores.
+var showLeaderboard = function (input) {
+  var output = "";
+  for (var i = 0; i < input.length; i += 1) {
+    output += `Player ${i + 1}: ${input[i]} <br>`;
+  }
+  return output;
+};
+
+// Variable to initialise while loop. Left out of main fuction so it does not reset to 0 everytime "submit" is clicked.
+var playerCount = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameVersion == "") {
+    return getVersion(input);
+  } else if (numPlayers == 0) {
+    return getPlayers(input);
+  } else if (numPlayers != 0 && numDice == 0) {
+    return getNumDice(input);
+  } else {
+    while (playerCount < numPlayers) {
+      if (gameMode == 1) {
```

Comment:
Good use of if and else ifs  

Usually when there are a number of else ifs we'll try to simplify by using if and return . Reason being that it makes the code more readable. Something like this
```
  if (gameVersion == "") {
    return getVersion(input);
  } 
  
  if (numPlayers == 0) {
    return getPlayers(input);
  } 
  
  if (numPlayers != 0 && numDice == 0) {
    return getNumDice(input);
  } else {
    while (playerCount < numPlayers) {
      if (gameMode == 1) {  ..................
```